custom/ilsgateway/filters.py

```diff
@@ -1,23 +1,55 @@
 from django.utils.translation import ugettext_noop
-from corehq.apps.products.models import Product
+from corehq.apps.products.models import SQLProduct
 from corehq.apps.programs.models import Program
 from corehq.apps.reports.filters.base import BaseDrilldownOptionFilter
 
 
 class ProductByProgramFilter(BaseDrilldownOptionFilter):
     slug = "filter_by"
+    single_option_select = 0
+    template = "opm/drilldown_options.html"
     label = ugettext_noop("Filter By")
 
     @property
     def drilldown_map(self):
-        options = []
+        options = [{"val": "0", "text": "All", "next": []}]
```

Comment:
I think you're just using "0" as a magic value to represent all data. If that's right it seems like changing it to `"all"` would be more clear (and probably good to make it a constant in this module)